ui/admin/tests/acceptance/credential-store/credentials/update-test.js

```diff
@@ -42,58 +46,150 @@ module(
       urls.projectScope = `/scopes/${instances.scopes.project.id}`;
       urls.credentialStores = `${urls.projectScope}/credential-stores`;
       urls.staticCredentialStore = `${urls.credentialStores}/${instances.staticCredentialStore.id}`;
       urls.credentials = `${urls.staticCredentialStore}/credentials`;
-      urls.credential = `${urls.credentials}/${instances.credential.id}`;
+      urls.usernamePasswordCredential = `${urls.credentials}/${instances.usernamePasswordCredential.id}`;
+      urls.usernameKeyPairCredential = `${urls.credentials}/${instances.usernameKeyPairCredential.id}`;
       authenticateSession({});
     });
 
-    test('can save changes to existing credential', async function (assert) {
+    test('can save changes to existing username & password credential', async function (assert) {
       assert.expect(3);
-      assert.notEqual(instances.credential.name, 'random string');
-      await visit(urls.credential);
+      assert.notEqual(
+        instances.usernamePasswordCredential.name,
+        'random string'
+      );
+      await visit(urls.usernamePasswordCredential);
       await click('form [type="button"]', 'Activate edit mode');
       await fillIn('[name="name"]', 'random string');
       await click('.rose-form-actions [type="submit"]');
-      assert.strictEqual(currentURL(), urls.credential);
+      assert.strictEqual(currentURL(), urls.usernamePasswordCredential);
       assert.strictEqual(
-        this.server.schema.credentials.all().models[0].name,
+        this.server.schema.credentials.where({ type: 'username_password' })
+          .models[0].name,
         'random string'
       );
     });
 
-    test('cannot make changes to an existing credential without proper authorization', async function (assert) {
+    test('can save changes to existing username & key pair credential', async function (assert) {
+      assert.expect(3);
+      assert.notEqual(
+        instances.usernameKeyPairCredential.name,
+        'random string'
+      );
+      await visit(urls.usernameKeyPairCredential);
+      await click('form [type="button"]', 'Activate edit mode');
+      await fillIn('[name="name"]', 'random string');
+      await click('.rose-form-actions [type="submit"]');
+      assert.strictEqual(currentURL(), urls.usernameKeyPairCredential);
+      assert.strictEqual(
+        this.server.schema.credentials.where({ type: 'ssh_private_key' })
+          .models[0].name,
+        'random string'
+      );
+    });
+
+    test('cannot make changes to an existing username & password credential without proper authorization', async function (assert) {
+      assert.expect(1);
+      instances.usernamePasswordCredential.authorized_actions =
+        instances.usernamePasswordCredential.authorized_actions.filter(
+          (item) => item !== 'update'
+        );
+      await visit(urls.usernamePasswordCredential);
+      assert.notOk(find('.rose-layout-page-actions .rose-button-secondary'));
```

Comment:
Can we update this assert to the below suggestion? Helps make it very clear what we are asserting here.
```suggestion
      assert.dom('.rose-layout-page-actions .rose-button-secondary').doesNotExist();
```